app/backend/src/couchers/rate_limits/definitions.py

```diff
@@ -22,24 +23,25 @@
 from couchers.utils import now
 
 if TYPE_CHECKING:
-    from sqlalchemy.orm import Session
+    pass
 
 
 @dataclass
 class RateLimitDefinition:
     warning_limit: int
     hard_limit: int
-    count_actions_query: Callable[["Session", int], int]
-    mod_email_information_query: Callable[["Session", int], list[dict]]
+    count_actions_query: Callable[[Session, int], int]
+    mod_email_information_query: Callable[[Session, int], list[dict[str, Any]]]
 
 
 RATE_LIMIT_INTERVAL = timedelta(hours=24)
 RATE_LIMIT_INTERVAL_STRING = "24 hours"
 
 
-def _get_user_host_requests_in_past_time_interval(session, user_id) -> list[dict]:
-    return (
-        session.execute(
+def _get_user_host_requests_in_past_time_interval(session: Session, user_id: int) -> list[dict[str, Any]]:
```

Comment:
I have a feeling the type was wrong here?

I don't think iterating through and applying `dict` does the same thing.